core/test_segment.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+FILE=test_segment.go
+
+pushd core
```

Comment:
This imposes an expectation that the script is run only from the parent directory, which is kind of weird. One way to mitigate this is just to take the output filename as an argument to the script, eg

FILE="$1"

Then in the makefile:

```
core/testSegment.go: core/test_segment.sh
    core/test_segment.sh core/testSegment.go
```

Another way is to use `dirname` within the script to change the working directory to the script's base path, eg:

```
cd $(dirname "0")
FILE=test_segment.go
```
No need for pushd / popd here; changes to the shell environment typically don't persist outside scripts.

Reply:
Added argument